ui/components/component-library/form-text-field/form-text-field.tsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import classnames from 'classnames';
+import {
+  Display,
+  FlexDirection,
+} from '../../../helpers/constants/design-system';
+import {
+  Box,
+  TextField,
+  HelpText,
+  HelpTextSeverity,
+  Label,
+  TextFieldSize,
+} from '..';
+import { PolymorphicRef } from '../box';
+import type { BoxProps } from '../box';
+import type { TextFieldProps } from '../text-field/text-field.types';
+import {
+  FormTextFieldSize,
+  FormTextFieldProps,
+  FormTextFieldComponent,
+} from './form-text-field.types';
+
+export const FormTextField: FormTextFieldComponent = React.forwardRef<
+  PolymorphicRef<any>,
+  FormTextFieldProps<any>
+>(
+  (
+    {
+      autoComplete,
+      autoFocus,
+      className = '',
+      defaultValue,
+      disabled,
+      isDisabled,
+      error,
+      helpText,
+      helpTextProps,
+      id,
+      inputProps,
+      inputRef,
+      label,
+      labelProps,
+      startAccessory,
+      maxLength,
+      name,
+      onBlur,
+      onChange,
+      onFocus,
+      placeholder,
+      readOnly,
+      required,
+      endAccessory,
+      size = FormTextFieldSize.Md,
+      textFieldProps,
+      truncate,
+      type = 'text',
+      value,
+      ...props
+    },
+    ref,
+  ) => {
+    // Validation function for 'id' prop
+    const validateIdProp = () => {
+      if (label && !id) {
+        throw new Error(
+          `If a label prop exists, you must provide an 'id' prop for the label's htmlFor attribute for accessibility. Warning coming from FormTextField component.`,
+        );
+      }
+    };
+
+    // Invoke the validation function
+    React.useEffect(() => {
+      validateIdProp();
+    }, [label, id]);
```

Comment:
Can we check if label is always used in the instances for this component and if so we could just make id a required prop? Otherwise I'll have to have a think about it when I get back tomorrow